pkg/router/template/router.go

```diff
@@ -616,8 +616,8 @@ func (r *templateRouter) dynamicallyAddRoute(backendKey string, route *routev1.R
 	newWeights := r.calculateServiceWeights(backend.ServiceUnits)
 	for key := range backend.ServiceUnits {
 		if service, ok := r.findMatchingServiceUnit(key); ok {
-			newEndpoints := service.EndpointTable
-			glog.V(4).Infof("For new route backend %s, replacing endpoints for service %s: %+v", backendKey, key, service.EndpointTable)
+			newEndpoints := endpointsForAlias(*backend, service)
```

Comment:
Not to reviewer: `endpointsForAlias` is defined in template_helper.go and intended to be called from the haproxy configuration template. It does exactly the job, but I'm not sure how fine it is to call it from outside the context of processing the template.

Reply:
That's ok as its a helper function in the same package.